packages/flutter_tools/lib/src/isolated/native_assets/macos/native_assets_host.dart

```diff
@@ -104,6 +114,126 @@ Future<void> setInstallNameDylib(File dylibFile) async {
   }
 }
 
+Future<Set<String>> getInstallNamesDylib(File dylibFile) async {
+  final ProcessResult installNameResult = await globals.processManager.run(
+    <String>[
+      'otool',
+      '-D',
+      dylibFile.path,
+    ],
+  );
+  if (installNameResult.exitCode != 0) {
+    throwToolExit(
+      'Failed to get the install name of $dylibFile:\n${installNameResult.stderr}',
+    );
+  }
+
+  // The output of `otool -D` looks like below. For each architecture, a
+  // separate install name is reported, which are not necessarily the same.
+  //
+  // /build/native_assets/ios/buz.framework/buz (architecture x86_64):
+  // @rpath/libbuz.dylib
+  // /build/native_assets/ios/buz.framework/buz (architecture arm64):
+  // @rpath/libbuz.dylib
+
+  final Iterator<String> lines =
+    (installNameResult.stdout as String).trim().split('\n').iterator;
+  final Set<String> installNames = <String>{};
+
+  while (lines.moveNext()) {
+    final String line = lines.current;
+    if (line.contains('(architecture')) {
+      continue;
+    }
+    installNames.add(line.trim());
+  }
+
+  return installNames;
+}
+
+Future<List<String>> getDependencyInstallNamesDylib(File dylibFile) async {
+  final ProcessResult listDependenciesResult = await globals.processManager.run(
+    <String>[
+      'otool',
+      '-L',
+      dylibFile.path,
+    ],
+  );
+  if (listDependenciesResult.exitCode != 0) {
+    throwToolExit(
+      'Failed to get the dependencies of $dylibFile:\n${listDependenciesResult.stderr}',
+    );
+  }
+
+  // The output of `otool -L` looks like below. For each architecture, there
```

Comment:
Could be moved into a separate function and have unit tests in `test/general.shard/isolated/macos/native_assets_host_test.dart`. (In case we run into future cases of `otool` that have slightly different output and want to ensure we keep parsing all versions.)